### PR TITLE
update monies-man bot to account for new notion DBs

### PR DESCRIPTION
**Pull request by rishitaro**
adjusting the name fetching logic for monies man

### CODE REVIEW

**Comment by rishitaro** [comment]
I'm merging this one myself too lol

**Comment by rishitaro** [comment]
ok nvm kevin's sending me secrets so now I feel obligated to test ✋

File: orders/OrderBot.ts [hunk 1]
```diff
@@ -42,7 +42,25 @@ export class OrderBot extends BaseSushiBot {
     }
 
     private getPropertyEmbed(data: any, propertyName: string): APIEmbedField | null {
-        const propertyValue = this.getTextPropertyValue(data, propertyName);
+        var propertyValue;
```
Comment (KTOmega) [inline orders/OrderBot.ts:45]: nit: use `let` instead
```suggestion
        let propertyValue;
```

File: orders/OrderFormNotion.ts [hunk 2]
```diff
@@ -36,26 +38,34 @@ export default class OrderForm {
...
+                    }
 
-                        this.bot.updateUsers(rosterEntry?.discordTag ?? null, i.properties);
+                    try {
+                        const submitterName = i.properties["Submitter"].people.at(0);
+                        if (submitterName != null && isFullUser(submitterName) && submitterName.name != null) {
+                            const rosterEntry = await this.notion.getRosterEntryFromName(submitterName.name);
+                            this.bot.updateUsers(rosterEntry?.discordTag ?? null, i);
+                        } else {
+                            throw Error("Submitter object malformed")
```
Comment (KTOmega) [inline orders/OrderFormNotion.ts:67]: nit: style (don't think the `new` does anything major for `Error`)

```suggestion
                            throw new Error("Submitter object malformed");
```

File: orders/OrderFormNotion.ts [hunk 3]
```diff
@@ -68,4 +78,14 @@ export default class OrderForm {
             console.warn(`Order form checker: An error occurred when checking for order form updates.`, e);
         }
     }
+
+    private completedSushiOrder(orderObject: PageObjectResponse) {
```
Comment (KTOmega) [inline orders/OrderFormNotion.ts:82]: nit: add return type
```suggestion
    private completedSushiOrder(orderObject: PageObjectResponse): boolean {
```

File: orders/OrderFormNotion.ts [hunk 4]
```diff
@@ -68,4 +78,14 @@ export default class OrderForm {
...
+        if (orderObject.properties["Order Description"]?.type == "title" && orderObject.properties["Product Name"]?.type == "rich_text" && orderObject.properties["Subtotal"]?.type == "number") {
+            const title = orderObject.properties["Order Description"]?.title.at(0)?.plain_text ?? null;
+            const subtotal = orderObject.properties["Subtotal"].number ?? null;
+
+            return title && title != "<Seller / Product Description> NOT your name" && subtotal;
```
Comment (KTOmega) [inline orders/OrderFormNotion.ts:87]: If TypeScript is happy to booleanify this then it's fine but maybe we can make it explicitly boolean:

```suggestion
            return title != "<Seller / Product Description> NOT your name" && subtotal != null;
```

File: orders/OrderFormNotion.ts [hunk 5]
```diff
@@ -68,4 +78,14 @@ export default class OrderForm {
             console.warn(`Order form checker: An error occurred when checking for order form updates.`, e);
         }
     }
+
+    private completedSushiOrder(orderObject: PageObjectResponse) {
+        if (orderObject.properties["Order Description"]?.type == "title" && orderObject.properties["Product Name"]?.type == "rich_text" && orderObject.properties["Subtotal"]?.type == "number") {
```
Comment (KTOmega) [inline orders/OrderFormNotion.ts:83]: minor: can we extract these props to consts? Similar to what we have in `NotionClient.ts`

Comment (KTOmega) [reply orders/OrderFormNotion.ts:83]: And maybe we can use those consts in `OrderBot.ts` too

**Review by KTOmega — APPROVED** [review]
A few minor nitpicks, like typings and stuff. Nothing major, I think it can be merged without these changes.

File: orders/OrderFormNotion.ts (hunk 5 above)
Comment (rishitaro) [reply orders/OrderFormNotion.ts:83]: I think there's some larger refactoring I can do! I'm going to do this and some more in a diff PR

**Comment by rishitaro** [comment]
thanks for taking a look @KTOmega, I'm going to merge for now to get the bot up and running and will do some refactoring in a New™ branch!

**Pull request merged**